File: codesignal/arcade/2.core/9.well of integration/4.alphabetSubsequence.py
# Check whether the given string is a subsequence of the plaintext alphabet.

class Solution:
    def alphabetSubsequence(s):
        return "".join(sorted(s)) == s and len(set(s)) == len(s)
    
class Solution:
    def alphabetSubsequence(s):
          for i in range(1, len(s)):
            if s[i] <= s[i-1]:
              return False
          return True

File: codesignal/arcade/2.core/9.well of integration/4.alphabetSubsequence.cpp
// Check whether the given string is a subsequence of the plaintext alphabet.

#include <string>
#include <algorithm>
using namespace std;

class Solution {
public:
  bool alphabetSubsequence(string s) {
    return is_sorted(s.begin(), s.end()) && adjacent_find(s.begin(), s.end()) == s.end();
  }
};

class Solution {
public:
  bool alphabetSubsequence(string s) {
    for (int m = 1; m < s.length(); m++) {
      if (s[m] <= s[m-1]) {
        return false;
      }
    }
    return true;
  }
};